crates/host/src/module.rs

```diff
@@ -138,38 +147,53 @@ pub const WASM_METERING_LIMIT: u64 = 100_000_000_000;
 /// We don't want tests to run forever, and it can take several minutes for 100 giga ops to run.
 pub const WASM_METERING_LIMIT: u64 = 10_000_000;
 
+#[cfg(feature = "wasmer_sys")]
 /// Generate an engine with a wasm compiler
 /// and Metering (use limits) in place.
-pub fn make_compiler_engine() -> Engine {
+pub fn make_engine() -> Engine {
     let cost_function = |_operator: &wasmparser::Operator| -> u64 { 1 };
     // @todo 100 giga-ops is totally arbitrary cutoff so we probably
     // want to make the limit configurable somehow.
     let metering = Arc::new(Metering::new(WASM_METERING_LIMIT, cost_function));
     // the only place where the wasm compiler engine is set
     let mut compiler = Cranelift::default();
     compiler.canonicalize_nans(true).push_middleware(metering);
-    Engine::from(compiler)
+
+    // Workaround for invalid memory access on iOS.
+    // https://github.com/holochain/holochain/issues/3096
+    let mut engine = Engine::from(compiler);
+    engine.set_tunables(BaseTunables {
+        static_memory_bound: 0x4000.into(),
+        static_memory_offset_guard_size: 0x1_0000,
+        dynamic_memory_offset_guard_size: 0x1_0000,
```

Comment:
Some comments would be awesome for these magic numbers.

Reply:
oh, i guess this wasn't new code you added. no worries if you don't have the context to add comments here.